plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -106,13 +104,16 @@ private void doFlushMetadataCache(ConnectorSession session, Optional<String> sch
         }
         else if (schemaName.isPresent() && tableName.isPresent()) {
             HiveMetastore metastore = metastoreFactory.createMetastore(Optional.of(session.getIdentity()));
-            Table table = metastore.getTable(schemaName.get(), tableName.get())
-                            .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(schemaName.get(), tableName.get())));
-            verifyDeltaLakeTable(table);
-            cachingHiveMetastore.ifPresent(caching -> caching.invalidateTable(table.getDatabaseName(), table.getTableName()));
-            String tableLocation = getTableLocation(table);
-            transactionLogAccess.invalidateCaches(tableLocation);
-            extendedStatisticsAccess.invalidateCache(tableLocation);
+            // This may insert into a cache, but this will get invalidated below. TODO fix Delta so that flush_metadata_cache doesn't have to read from metastore
+            Optional<Table> tableBeforeFlush = metastore.getTable(schemaName.get(), tableName.get());
+            cachingHiveMetastore.ifPresent(caching -> caching.invalidateTable(schemaName.get(), tableName.get()));
+
+            Optional<String> tableLocation = tableBeforeFlush.map(table -> {
+                verifyDeltaLakeTable(table);
+                return getTableLocation(table);
+            });
+            tableLocation.ifPresent(transactionLogAccess::invalidateCaches);
+            tableLocation.ifPresent(extendedStatisticsAccess::invalidateCache);
```

Comment:
Maybe the cache key for these should be a `tableName, tableLocation` tuple?

Reply:
yes, i think so

Reply:
i added `TODO fix Delta so that flush_metadata_cache doesn't have to read from metastore` for now.
would prefer to revisit this after https://github.com/trinodb/trino/pull/17092

Reply:
-> https://github.com/trinodb/trino/pull/17214